src/libs/ReportUtils.ts

```diff
@@ -3317,6 +3317,157 @@ function getIcons(
     return getIconsForParticipants(participantAccountIDs, personalDetails);
 }
 
+function getReportActionAvatars({
+    iouReport,
+    action,
+    chatReport,
+    personalDetails,
+    policies,
+    reportPreviewSenderID,
+
+    shouldStackHorizontally = false,
+    shouldUseCardFeed = false,
+    isReportArchived = false,
+    accountIDs = [],
+}: {
+    iouReport: OnyxEntry<Report>;
+    action: OnyxEntry<ReportAction>;
+    chatReport: OnyxEntry<Report>;
+    personalDetails: OnyxEntry<PersonalDetailsList>;
+    policies: OnyxCollection<Policy>;
+    reportPreviewSenderID: number | undefined;
+
+    shouldStackHorizontally?: boolean;
+    shouldUseCardFeed?: boolean;
+    isReportArchived?: boolean;
+    accountIDs?: number[];
+}) {
+    /* Get avatar type */
+
+    const policyID = chatReport?.policyID === CONST.POLICY.ID_FAKE || !chatReport?.policyID ? (iouReport?.policyID ?? chatReport?.policyID) : chatReport?.policyID;
+    const policy = policies?.[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`];
+
+    const isATripRoom = isTripRoom(chatReport);
+    const isWorkspaceWithoutChatReportProp = !chatReport && policy?.type !== CONST.POLICY.TYPE.PERSONAL;
+    const isAWorkspaceChat = isPolicyExpenseChat(chatReport) || isWorkspaceWithoutChatReportProp;
+    const isATripPreview = action?.actionName === CONST.REPORT.ACTIONS.TYPE.TRIP_PREVIEW;
+    const isAReportPreviewAction = action?.actionName === CONST.REPORT.ACTIONS.TYPE.REPORT_PREVIEW;
+    const isReportPreviewOrNoAction = !action || isAReportPreviewAction;
+    const isReportPreviewInTripRoom = isAReportPreviewAction && isATripRoom;
+
+    // We want to display only the sender's avatar next to the report preview if it only contains one person's expenses.
+    const displayAllActors = isAReportPreviewAction && !isATripRoom && !isAWorkspaceChat && !reportPreviewSenderID;
+
+    const shouldUseAccountIDs = accountIDs.length > 0;
+    const shouldShowAllActors = displayAllActors && !reportPreviewSenderID;
+    const isChatThreadOutsideTripRoom = isChatThread(chatReport) && !isATripRoom;
+    const shouldShowSubscriptAvatar = shouldReportShowSubscript(iouReport ?? chatReport, isReportArchived) && policy?.type !== CONST.POLICY.TYPE.PERSONAL;
```

Comment:
I wonder if these should be utils helper functions. We're doing multiple calculations to decide the type of avatar both here and in the ReportActionAvatars folder. WDYT?

NAB